### ISSUE TITLE
New info file items

### ISSUE THREAD

**Issue by indigoxela** [backdrop-contrib/info_tester_theme#1]
Hey there, so you're testing stuff in combination with b-org... And I do have some questions (sorry for posting here, but it seemed appropriate for me). @docwilmot feel free to point me to a better place.

```
; Themes can have colors, tags, maintainers, and screenshots.

; Aquamarine shouldnt work, since its not in the allowed list.
colors[] = Blue
colors[] = Beige
colors[] = Aquamarine
```
Colors? What about Color module support? These themes can have all sorts of colors. And honestly, color would never be a criterion for me when searching for nice themes.

Or is this just about testing some random things out?

```
tags[] = Testing
tags[] = Development
```
Same as in modules? For modules these tags also provide the base for  grouping on page /admin/modules/list - is something like that also considered for themes?
Or are tags for themes only considered for search (on b-org and in the installer)? In this case tags mean something different for modules and themes.
(It might be hard to find accurate tags for themes, I guess.)

```
maintainers[] = docwilmot
maintainers[] = bugfolder
```
Is this to display the maintainer on the b-org page? Right now that info is supposed to be in the readme. What's the benefit of an extra info item? Is that easier to use when displaying on b-org? Or is this for a different purpose? Search?

```
; Screenshot handling rules
screenshots[default] = firstpng.png
```
Now I'm really curious. Until now all themes are supposed to have one file named "screenshot.png" in the main directory. That one's also used in the display on page /admin/appearance/list
Will this change in the future?

Are multiple images possible?
Do these also work for modules? For modules it's currently a bit tricky to provide a (UI) screenshot.

**Comment by docwilmot** [comment]
This module, and the issue is came from https://github.com/backdrop-ops/backdropcms.org/issues/487 is to try and get more info about projects into BackdropCMS.org. D7 hosted projects on their own platform so info about projects and maintainers was searchable, but because we use GitHub, that info isnt as easy to get to. So we're trying to allow info files to store more data. When a project on GitHub is released, the data from the info file is stored in the project node there as fields. Currently we have fields on the project nodes for :
- colors (themes only, a defined list of colors)
- screenshots (all projects)
- maintainers (all projects)
- tags (all projects)

This will allow users on BackdropCMS.org to search for projects by (theme) color, developer and tags; and to see screenshots of all projects. 

So far in our recent testing, this seems to be working. So next step is to figure ways to display these fields on BackdropCMS.org in the project listings. Design is not my forte, so I'm hoping others have ideas. Another step is to let devs know that they can add this info to their projects now.
> And honestly, color would never be a criterion for me when searching for nice themes

Probably not, but for now its just a field on the node, if its not needed, we dont need to use it.
> Until now all themes are supposed to have one file named "screenshot.png" in the main director

We're aiming to enhance this. Since we can now add a screenshots folder, it seems unnecessary to still *require* a screenshot outside that folder, so the code we just tested will look for a screenshot if one is not in the root. It checks in order:
- is there a screenshot.png in root (this is existing behavior)
- is there a `screenshot = location/name.png` in the info file (this is existing behavior)
- is there a `screenshot.png` in the folder
- random image from the folder

**Comment by indigoxela** [comment]
> Since we can now add a screenshots folder, it seems unnecessary to still require a screenshot outside that folder,

Keep in mind, that this might also need changes in core's system module: https://github.com/backdrop/backdrop/blob/835438c1c59d6fdae626f10a143aae9a089ce1b9/core/modules/system/system.module#L3087

Screenshots being available for modules the same way they're for themes, is a great enhancement.

I still have concerns regarding "tags" - they do different things for modules and for themes, if I get it right. How about introducing something new like "keywords" - only for online searching. Or would this get even more confusing?

And I still find "colors" for themes pretty... ehm...
Maybe ask some actual themers, what they'd find useful. :wink:

Anyway, great that now there's some progress. I know, the sparse information in info files is one of your pet peeves for quite a while.

**Comment by docwilmot** [comment]
Yes, to fully utilize this, we'll need to make changes also in core and backdropcms.
Tags are just labels, no different on theme, layout or module. only to categorize the project. Initially to allow searching on BackdropCMS, maybe eventually for use in core.
> Maybe ask some actual themers, what they'd find useful

I think we somehow did: this issue has been around and mentioned repeatedly for 6 years. I suspect the thinking behind it would be that if we had a significant offering of themes and a non-expert user wanted something blue, he could filter by color. It certainly cant harm. 

What would you suggest would be ideal search filters for projects on BackdropCMS.org?

**Comment by indigoxela** [comment]
> Tags are just labels, no different on theme, layout or module

One small difference is that tags are also "searchable" on /admin/modules/list, but maybe that's OK.

> What would you suggest would be ideal search filters for projects

If I knew, I would have posted in the backdrop-ops issue. :wink: I've no idea.
